app/tests/algorithms_tests/test_views.py

```diff
@@ -2018,3 +2020,57 @@ def test_update_view_limits_gpu_choice(client):
     )
 
     assert response.status_code == 302
+
+
+@pytest.mark.django_db
+def test_algorithm_template_download(client):
```

Comment:
Think AAA. Probably better to split this into two tests - one for permissions, one for functionality.

Reply:
Looked at it afresh: turns out there was already a collection of 'check permission vs views' in the algorithms. I've added the permission checking to that.